File: src/OrbitShield.Application/Mission/MissionDtos.cs
namespace OrbitShield.Application.Mission;

public sealed record TelemetryResponse(
    int SatelliteId,
    string SatelliteCode,
    int Fuel,
    int SolarEnergy,
    string OrbitStatus);

public sealed record ConjunctionResponse(
    bool CollisionRisk,
    decimal Distance,
    decimal Probability,
    string DebrisId,
    string Status);

public sealed record ManeuverRequest(
    int SatelliteId,
    int? ConjunctionEventId,
    string Action,
    int ServoAngle,
    decimal? ThrustLevel,
    string Source);

public sealed record ManeuverResponse(bool Accepted, string Message);

public sealed record ManeuverLogResponse(
    int Id,
    int SatelliteId,
    int? ConjunctionEventId,
    string Action,
    int ServoAngle,
    decimal? ThrustLevel,
    string Source,
    DateTime ExecutedAt);

public sealed record SensorReadingRequest(
    int SatelliteId,
    decimal TemperatureCelsius,
    decimal RadiationLevel,
    decimal BatteryVoltage,
    decimal? SimulatedGravity,
    decimal? SimulatedThrust);

public sealed record SensorReadingResponse(
    int Id,
    int SatelliteId,
    decimal TemperatureCelsius,
    decimal RadiationLevel,
    decimal BatteryVoltage,
    decimal? SimulatedGravity,
    decimal? SimulatedThrust,
    DateTime CreatedAt);

public sealed record OrbitalElementRequest(
    int SatelliteId,
    string Name,
    string TleLine1,
    string TleLine2,
    DateTime Epoch,
    string Source);

public sealed record OrbitalElementResponse(
    int Id,
    int SatelliteId,
    string Name,
    string TleLine1,
    string TleLine2,
    DateTime Epoch,
    string Source,
    DateTime CreatedAt);

File: android/OrbitShieldEngineer/app/src/main/java/com/orbitshield/engineer/data/remote/dto/MissionDtos.kt
package com.orbitshield.engineer.data.remote.dto

data class TelemetryDto(
    val satelliteId: Int,
    val satelliteCode: String,
    val fuel: Int,
    val solarEnergy: Int,
    val orbitStatus: String
)

data class SensorReadingDto(
    val id: Int,
    val satelliteId: Int,
    val temperatureCelsius: Double,
    val radiationLevel: Double,
    val batteryVoltage: Double,
    val simulatedGravity: Double?,
    val simulatedThrust: Double?,
    val createdAt: String
)

data class ManeuverLogDto(
    val id: Int,
    val satelliteId: Int,
    val conjunctionEventId: Int?,
    val action: String,
    val servoAngle: Int,
    val thrustLevel: Double?,
    val source: String,
    val executedAt: String
)

data class OrbitalVectorDto(
    val x: Double,
    val y: Double,
    val z: Double
)

data class OrbitalEnvironmentDto(
    val satelliteId: Int,
    val satelliteCode: String,
    val source: String,
    val tleName: String,
    val tleLine1: String,
    val tleLine2: String,
    val epochUtc: String,
    val satellitePositionKm: OrbitalVectorDto,
    val satelliteVelocityKmS: OrbitalVectorDto,
    val relativePositionKm: OrbitalVectorDto,
    val relativeVelocityKmS: OrbitalVectorDto,
    val safeDistanceKm: Double,
    val lookaheadSeconds: Double,
    val timeToClosestApproachSeconds: Double,
    val missDistanceKm: Double,
    val relativeSpeedKmS: Double,
    val debrisDiameterMeters: Double,
    val estimatedMassKg: Double,
    val impactEnergyJoules: Double,
    val debrisClass: String,
    val scenarioClassification: String,
    val collisionProbability: Double,
    val probabilityModel: String,
    val recommendedEmergency: Boolean,
    val predictedImpact: Boolean
)
